compiler/src/function/input/main_function_input.rs

```diff
@@ -58,7 +60,82 @@ impl<'a, F: PrimeField, G: GroupType<F>> ConstrainedProgram<'a, F, G> {
             )?)),
             Type::Array(type_, len) => self.allocate_array(cs, name, &*type_, *len, input_option, span),
             Type::Tuple(types) => self.allocate_tuple(cs, &name, types, input_option, span),
-            _ => unimplemented!("main function input not implemented for type"),
+            _ => unimplemented!("main function input not implemented for type {}", type_),
+        }
+    }
+}
+
+/// Process constant inputs and return ConstrainedValue with constant value in it.
+impl<'a, F: PrimeField, G: GroupType<F>> ConstrainedProgram<'a, F, G> {
+    pub fn constant_main_function_input<CS: ConstraintSystem<F>>(
+        &mut self,
+        _cs: &mut CS,
+        type_: &Type,
+        name: &str,
+        input_option: Option<InputValue>,
+        span: &Span,
+    ) -> Result<ConstrainedValue<'a, F, G>, FunctionError> {
+        let input = input_option.ok_or_else(|| FunctionError::input_not_found(name.to_string(), span))?;
+
+        match (type_, input) {
+            (Type::Address, InputValue::Address(addr)) => Ok(ConstrainedValue::Address(Address::constant(addr, span)?)),
+            (Type::Boolean, InputValue::Boolean(value)) => Ok(ConstrainedValue::Boolean(Boolean::constant(value))),
+            (Type::Field, InputValue::Field(value)) => Ok(ConstrainedValue::Field(FieldType::constant(value, span)?)),
+            (Type::Group, InputValue::Group(value)) => Ok(ConstrainedValue::Group(G::constant(&value.into(), span)?)),
+            (Type::Integer(integer_type), InputValue::Integer(_, value)) => {
+                let const_int = match integer_type {
+                    IntegerType::U8 => ConstInt::U8(value.parse::<u8>().unwrap()),
```

Comment:
Is there a way we can return the parsing errors instead of unwrap?

Reply:
Yeah. ConstInt supports parsing. Fixed.